### PR TITLE
docs: Add v2022.1 release notes

### PR DESCRIPTION
**Pull request by blocktrron**
This is still unfinished and will most likely not render correctly in ReadTheDocs.

But let's have a first cycle so we can push this through very soon.

Contains changes up to b8513546

### CODE REVIEW

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:349]: - `null@l2tp` needs to be at the **top** of the list
- `mesh-vpn-fastd-l2tp` needs to be added within the site.mk

Maybe it's better to just link to https://github.com/freifunk-gluon/gluon/blob/master/docs/features/vpn.rst#methods. idk

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:194]: sorting necessary?

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:274]: it's no drop-in replacement for fastd though, is it? if not, we should mention the relevant difference to current fastd setups

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:342]: "deriavation" has a spelling error. and wouldn't "variant" be a better word here?

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:344]: "is not used" ? i think this is not correct?

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:345]: independant -> independent

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:357]: existant -> existent

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [reply docs/releases/v2022.1.rst:274]: Open for debate I suppose,
but while WireGuard on it's own comes short regarding our needs to fastd due to L3 only,
the current implementation in gluon using VXLAN and the wgpeerselector very much is a drop in replacement.

In fact even the changes in the siteconf are minimal.
The biggest issue will be setting up the gateways accordingly.
Whoever wants to get that done instead of puzzling it out on his own is invited to take a look at our [ansible roles](https://github.com/freifunkh/ansible/tree/master/roles/ffh.mesh_wireguard).

Both 'can' even be run at the same time on the same machines, though that's not officially supported by gluon, requiring a split site-conf.

**Review by herbetom — CHANGES_REQUESTED** [review]
once #2611 and #2584 are merged the release notes probably should reference those sections

File: docs/releases/v2022.1.rst
Comment (T-X) [reply docs/releases/v2022.1.rst:296]: Now it sounds a bit like Wireguard is for performance, but fastd L2TP is not? Would it make sense to add a common paragraph to say "Gluon got two new VPN methods which have increased performance: Wireguard and fastd L2TP. Bc. both fully in kernel space etc." And then sub sections which describe their differences?

Also, currently the Wireguard section mentions VPN only at the end. And it might sound as if were increasing performance for any encrypted connection?

**Comment by T-X** [comment]
Should the removal of the bridge multicast wakeup call feature and potential downsides for Android users be mentioned?

**Comment by T-X** [comment]
There should be a mention that this Gluon release now uses OpenWrt 22.03.0-rc6+.

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:296]: I've fixed the link to the respective wiki page

@T-X This sounds like an improvement to the VPN wiki page and not the release notes, so I don't think this is actionable for this PR?

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:198]: To my knowledge, there's the Bullet / Nanostation family and which additional devices? I do not have a comprehensive overview about the cross-compatibility of these images.

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:194]: done

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:274]: @rotanid I do not have the expertise to write a comprehensive migration guide, so i see this as out of scope for the release-notes. We did not do this for L2TP using tunneldigger either. So maybe this is worth an additional wiki page, not the release-notes.

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:342]: Expanded to full list of affected chipsets

**Comment by blocktrron** [comment]
@T-X I've included the Bridge-Wakeup regression as "Known Bug", can you have a look?

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst]: Forgot the Picostation and Mixed up XM / XW. Is this now complete?

Comment (rotanid) [reply docs/releases/v2022.1.rst]: i think this looks better, only that the Rocket is not removed due to too little flash, the PR is simply missing its final tests, see #2580 , so this device would need to be mentioned in the other section of the RN

**Comment by AiyionPrime** [comment]
We talked about how we'd warn in the docs about the lacking WG key migration [in the last meetup](https://github.com/freifunk-gluon/gluon/wiki/Meeting-2022-04#discussion-about-the-v20221) in case we didn't merge #2601 in time.

@blocktrron Do you want that in the known bugs section too? Or somewhere else?

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [reply docs/releases/v2022.1.rst]: True, this should be in `Missing devices` as #2580 is currently just waiting for an LED confirmation...

**Comment by blocktrron** [comment]
@AiyionPrime key migration is a feature, neither a bug nor a regression

**Comment by AiyionPrime** [comment]
So somewhere else. Where?

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:286]:
```suggestion
WireGuard, a key derivation procedure is `currently being
developed <https://github.com/freifunk-gluon/gluon/pull/2601>`__. This
```

Comment (AiyionPrime) [reply docs/releases/v2022.1.rst:286]: it currently renders twice, once as text, once as link.

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:366]:
```suggestion
   configured to use a non-existent / invalid branch
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:372]:
```suggestion
  potentially breaking IPv6 connectivity for these devices after extended sleep periods
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:348]:
```suggestion
   ``bcm27xx-bcm2710``
```

**Review by AiyionPrime — CHANGES_REQUESTED** [review]
Looks really nice by now. Only minor stuff from my side;
I have now opinion on whether or not to use points at the end of sentences in listings, but maybe let's make it consistent.

The rocket should be marked in the right category and that doubling of `currently being developed` should be gone in some way.

File: docs/releases/v2022.1.rst
Comment (Grotax) [inline docs/releases/v2022.1.rst:1]: This should be 2022.1 right?

File: docs/releases/v2022.1.rst
Comment (rotanid) [inline docs/releases/v2022.1.rst:1]:
```suggestion
Gluon 2022.1
```

File: docs/releases/v2022.1.rst
Comment (rotanid) [reply docs/releases/v2022.1.rst]: fixed

File: docs/releases/v2022.1.rst
Comment (rotanid) [reply docs/releases/v2022.1.rst]: > 20:39:00 <               T_X > blocktrr1: thanks, the "Known issues" section - and everything else in the release notes, sounds good to me!
> 20:39:30 <               T_X > also thanks for adding a note for the new OpenWrt / Linux kernel / wireless backports version in this new release
>

**Comment by rotanid** [comment]
i worked through the open remarks.
i also added R6020 which was added to master branch in the meantime.

one thing i'm wondering though:
have you removed the old known issues from this RN on purpose? @blocktrron

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:347]:
```suggestion

Preconfigured Interfaces Roles
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Instead of ``mesh_on_wan`` and ``mesh_on_lan`` there is now a
``interfaces`` block to configure the default behavior of network
interfaces. For details take a look
:ref:`here<user-site-interfaces>`.

```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:335]: I tried to propose a link but without rebasing this PR it just doesn't make sense. But here should be the corresponding docs section which is now merged.

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:357]:
```suggestion
   radios (`#2541 <https://github.com/freifunk-gluon/gluon/pull/2541>`__)
```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:356]: this long line doesn't allow for line breaks

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:346]:
```suggestion
moved to ``mesh_vpn.fastd.mtu``. (`#2352 <https://github.com/freifunk-gluon/gluon/pull/2352>`__)
```
otherwise it renders for me to ([(#2352)](https://github.com/freifunk-gluon/gluon/pull/2352))

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:335]: now you can

File: docs/releases/v2022.1.rst
Comment (blocktrron) [reply docs/releases/v2022.1.rst:347]: Reworded

**Comment by blocktrron** [comment]
Added release notes and incorporated suggestions.

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:391]:
```suggestion
* A workaround for Android devices not waking up to their MLD subscriptions was removed,
  potentially breaking IPv6 connectivity for these devices after extended sleep periods
* Upgrading EdgeRouter-X from versions before v2020.1.x may lead to a soft-bricked state due to bad blocks on the NAND flash which the NAND driver before this release does not handle well.
```

duplicate "Known issues" title

File: docs/releases/v2022.1.rst
Comment (herbetom) [reply docs/releases/v2022.1.rst:335]: proposing changes to other files is to my knowledge not possible so: https://github.com/blocktrron/gluon/pull/9

**Comment by herbetom** [comment]
The changes from b85135467be94c280c3cdd9f4fe0adce3f928fe6 onward shouldn't be forgotten.

https://github.com/freifunk-gluon/gluon/compare/b85135467be94c280c3cdd9f4fe0adce3f928fe6...master

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:3]: Typically there would be a `Important  Notes` section at the beginnging.

https://gluon.readthedocs.io/en/latest/releases/v2021.1.html

Something like "Upgrades to v2022.1 and later releases are only supported from releases ..."

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:302]:
```suggestion
requires further modifications to a site :ref:`VPN fastd methods`.
```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:78]:
```suggestion
   -  WAC104
   -  WAX202
```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:312]:
```suggestion
In addition to a sufficiently
:ref:`configured fastd-based VPN server<vpn-gateway-configuration>`,
this requires further modifications to a site :ref:`VPN fastd methods`.
```

File: docs/releases/v2022.1.rst
Comment (maurerle) [inline docs/releases/v2022.1.rst:261]: Support was just (re)added for WNDR3800
Can be removed from list of not migrated devices.

Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:261]:
```suggestion
```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:118]:
```suggestion
   -  R6020
```
missing second space

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:84]:
```suggestion
   -  RE650 v1
```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:323]:
```suggestion
The ``ramips-mt7621`` and ``lantiq-xrx200`` targets now use the upstream DSA
```
uniformity

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:304]:
```suggestion
tunneling within Gluons mesh-vpn framework. This new ``null@l2tp`` 
connection Method allows for increased performance within existing
fastd setups.
```

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:352]:
```suggestion
moved to ``mesh_vpn.fastd.mtu``. (`#2352 <https://github.com/freifunk-gluon/gluon/pull/2352>`__)
```

idk, possible that i broke that myself with a suggestion

File: docs/releases/v2022.1.rst
Comment (herbetom) [inline docs/releases/v2022.1.rst:372]:
```suggestion
   radios (`#2541 <https://github.com/freifunk-gluon/gluon/pull/2541>`__)
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:116]:
```suggestion
-  Netgear
```

Comment (AiyionPrime) [reply docs/releases/v2022.1.rst:116]: indentation is wrong

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:269]:
```suggestion
-  Ubiquiti
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:286]:
```suggestion
WireGuard
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:305]:
```suggestion
connection method allows for increased performance within existing
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:310]:
```suggestion
this requires further modifications to a sites :ref:`VPN fastd methods<VPN fastd methods>`.
```

rendered as `[...] modifications to a site Method.`
now renders: `[...] modifications to a sites fastd methods.`

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:359]:
```suggestion
Instead of ``mesh_on_wan`` and ``mesh_on_lan`` there is now an
```

File: docs/releases/v2022.1.rst
Comment (AiyionPrime) [inline docs/releases/v2022.1.rst:375]:
```suggestion
-  WolfSSL instead of OpenSSL is now used when built with WPA3 support
```

the other bulletpoints in this section do not have dots either

**Review by AiyionPrime — CHANGES_REQUESTED** [review]
Minor fixes, reads good overall.
Have not looked into whether something is missing though.

**Comment by T-X** [comment]
Just noticed one more thing for the release, https://gluon.readthedocs.io/en/master/user/getting_started.html#building-the-images, there the "Replace RELEASE with the version you’d like to checkout, e.g. v2021.1." might need an update, too?

**Comment by herbetom** [comment]
@T-X it's part of the relase proccess to change that https://github.com/freifunk-gluon/gluon/wiki/Release-Process

**Pull request merged**
